[GPU] Add CMake option for incremental oneDNN builds (#36696)
openvinotoolkit/openvino: src/plugins/intel_gpu/thirdparty/CMakeLists.txt

        set(ONEDNN_BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/onednn_gpu_build")
        set(ONEDNN_INSTALL_DIR "${CMAKE_CURRENT_BINARY_DIR}/onednn_gpu_install" CACHE PATH "Installation path for oneDNN GPU library")
        set(ONEDNN_PREFIX_DIR "${CMAKE_CURRENT_BINARY_DIR}/onednn_gpu_root")

>         set(OV_ONEDNN_GPU_BUILD_ALWAYS OFF CACHE BOOL "Always run oneDNN GPU external build step (useful for local development)")

Reviewer: ```suggestion
        set(GPU_ONEDNN_BUILD_ALWAYS OFF CACHE BOOL "Always run oneDNN GPU external build step. It allows incremental build of oneDNN when it is changed.")
```
